api/src/main/java/org/openmrs/ChemoDosingInstructions.java

```diff
@@ -0,0 +1,171 @@
+			dosingInstructions.append(this.dosageAdjustmentPercentage);
+			dosingInstructions.append("\n");
+		}
+		if (this.dosageDeliveredTiming != null) {
+			dosingInstructions.append("Dosage Delivered Timing : ");
+			dosingInstructions.append(this.dosageDeliveredTiming);
+			dosingInstructions.append("\n");
+		}
+		if (this.dosageDilutionInstructions != null) {
+			dosingInstructions.append("Dosage Dilution Instructions : ");
+			dosingInstructions.append(this.dosageDilutionInstructions);
+			dosingInstructions.append("\n");
+		}
+		return dosingInstructions.toString();
+	}
+
+	/**
+	 * @see DosingInstructions#setDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public void setDosingInstructions(DrugOrder order) {
+
+		// storing dosing instruction type as "ChemoDosingInstructions"
+		order.setDosingType(this.getClass());
+
+		// storing dosing instruction string into DrugOrder object
+		order.setDosingInstructions(this.getDosingInstructionsAsString(Locale.getDefault()));
+	}
+
+	/**
+	 * @see DosingInstructions#getDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public DosingInstructions getDosingInstructions(DrugOrder order) {
+		if (!order.getDosingType().equals(this.getClass())) {
+			throw new APIException("DrugOrder.error.dosingTypeIsMismatched", new Object[] { this.getClass().getName(),
+				order.getDosingType() });
+		}
+		ChemoDosingInstructions chemoDosingInstructions = new ChemoDosingInstructions();
+
+		// dosing instructions in drug order consist of several lines each having a different type of instruction
+		StringTokenizer tokenizeDosingInstructions = new StringTokenizer(order.getDosingInstructions(), "\n");
+		while (tokenizeDosingInstructions.hasMoreTokens()) {
+
+			//Every instruction line is of the form "instructionLineType : instructionLineValue"
+			String instructionLine = tokenizeDosingInstructions.nextToken();
+			int indexOfColon = instructionLine.indexOf(':');
+			String instructionLineType = instructionLine.substring(0, indexOfColon - 1);
+			String instructionLineValue = instructionLine.substring(indexOfColon + 2);
+			if (instructionLineType.equals("Dosage Adjustment Percentage")) {
+				chemoDosingInstructions.dosageAdjustmentPercentage = Double.parseDouble(instructionLineValue);
+			} else if (instructionLineType.equals("Dosage Delivered Timing")) {
+				chemoDosingInstructions.dosageDeliveredTiming = instructionLineValue;
+			} else {
+				chemoDosingInstructions.dosageDilutionInstructions = instructionLineValue;
+			}
+		}
+		return chemoDosingInstructions;
+	}
+
+	/**
+	 * @see DosingInstructions#validate(DrugOrder, org.springframework.validation.Errors)
+	 */
+	@Override
+	public void validate(DrugOrder order, Errors errors) {
+		ValidationUtils.rejectIfEmpty(errors, "dosingInstructions",
+			"DrugOrder.error.dosingInstructionsIsNullForDosingTypeChemo");
+	}
+
+	/**
+	 * @see DosingInstructions#getAutoExpireDate(DrugOrder)
+	 */
+	@Override
+	public Date getAutoExpireDate(DrugOrder order) {
+		return null;
```

Comment:
Is this suppose to be null ?

Reply:
This method returns null if the auto-expire date cannot be inferred.
In this case, I cannot infer the auto-expire date from just the dosing instructions in drug order, hence it returns null.